tests/utils/interceptors.py

```diff
@@ -0,0 +1,33 @@
+from aidial_sdk.chat_completion.chunks import ContentChunk
+from typing_extensions import override
+
+from aidial_interceptors_sdk.chat_completion.base import (
+    ChatCompletionInterceptor,
+)
+
+
+class TracingInterceptor(ChatCompletionInterceptor):
+    # Interceptor parameter
+    name: str
+
+    traced_entered = False
+
+    def _send_line(self, content: str):
+        self.send_chunk(ContentChunk(content, 0))
+
+    def _trace_enter(self):
+        if not self.traced_entered:
+            self.traced_entered = True
+            self._send_line(f"{self.name}.enter\n")
+
+    def _trace_exit(self):
+        self._send_line(f"\n{self.name}.exit")
+
+    @override
+    async def on_stream_chunk(self, chunk: dict) -> None:
+        self._trace_enter()
```

Comment:
That's a valid way to achieve the roughly same effect, but I don't see why it's better than the given one.

Reply:
This comment was inspired by the idea of moving this interceptor to examples because there is no point in creating your own function `self._trace_enter()` when we have a dedicated method for such cases in sdk itself

If we think that's not worth doing, then there is no difference indeed